qiskit/qobj/pulse_qobj.py

```diff
@@ -0,0 +1,578 @@
+        """Instantiate a PulseQobjExperimentConfig object.
+
+        Args:
+            qubit_lo_freq (list): List of frequencies (as floats) for the qubit
+                driver LO's in GHz.
+            meas_lo_freq (list): List of frequencies (as floats) for the'
+                measurement driver LO's in GHz.
+            kwargs: Additional free form key value fields to add to the
+                configuration
+        """
+        self._data = {}
+        if qubit_lo_freq is not None:
+            self._data['qubit_lo_freq'] = qubit_lo_freq
+        if meas_lo_freq is not None:
+            self._data['meas_lo_freq'] = meas_lo_freq
+        if kwargs:
+            self._data.update(kwargs)
+
+
+class PulseLibraryItem:
+    """An item in a pulse library."""
+
+    def __init__(self, name, samples):
+        """Instantiate a pulse library item.
+
+        Args:
+            name (str): A name for the pulse.
+            samples (list[complex]): A list of complex values defining pulse
+                shape.
+        """
+        self.name = name
+        self.samples = samples
+
+    def to_dict(self):
+        """Return a dictionary format representation of the pulse library item.
+
+        Returns:
+            dict: The dictionary form of the PulseLibraryItem.
+        """
+        return {'name': self.name, 'samples': self.samples}
+
+    @classmethod
+    def from_dict(cls, data):
+        """Create a new PulseLibraryItem object from a dictionary.
+
+        Args:
+            data (dict): A dictionary for the experiment config
+
+        Returns:
+            PulseLibraryItem: The object from the input dictionary.
+        """
+        return cls(**data)
+
+    def __eq__(self, other):
+        if isinstance(other, PulseLibraryItem):
+            if self.to_dict() == other.to_dict():
+                return True
+        return False
+
+
+class PulseQobj:
+    """A Pulse Qobj."""
+
+    def __init__(self, qobj_id, config, experiments,
+                 header=None):
+        """Instatiate a new QASM Qobj Object.
+
+        Each QASM Qobj object is used to represent a single payload that will
+        be passed to a Qiskit provider. It mirrors the Qobj the published
+        `Qobj specification <https://arxiv.org/abs/1809.03452>`_ for OpenQASM
+        experiments.
+
+        Args:
+            qobj_id (str): An identifier for the qobj
+            config (PulseQobjConfig): A config for the entire run
+            header (QobjHeader): A header for the entire run
+            experiments (list): A list of lists of :class:`PulseQobjExperiment`
+                objects representing an experiment
+        """
+        self.qobj_id = qobj_id
+        self.config = config
+        self.header = header or QobjHeader()
+        self.experiments = experiments
+        self.type = 'PULSE'
+        self.schema_version = '1.1.0'
+
+    def _validate_json_schema(self, out_dict):
+        class PulseQobjEncoder(json.JSONEncoder):
```

Comment:
It seems that it'd be useful to have the encoder class in `utils`, so other providers can also use it to serialize Qobj if they choose to.

Reply:
I specifically didn't want to do this because the only thing which should ever need the custom encoder is the ibmq provider. No other provider out there is using qobj as the api interface it's strictly the iqx api. This is just the start of making that explicit.

That's a big part of why https://github.com/Qiskit/rfcs/pull/2 we're changing the interface to just pass terra objects.